tests/v2/validation/rbac/globalrolesv2/globalroles_v2_test.go

```diff
@@ -0,0 +1,501 @@
+package globalrolesv2
+
+import (
+	"testing"
+	"time"
+
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/rbac"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioning"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type GlobalRolesV2TestSuite struct {
+	suite.Suite
+	client       *rancher.Client
+	session      *session.Session
+	clusterCount int
+	user         *management.User
+	grbName      string
+	grName       string
+}
+
+func (gr *GlobalRolesV2TestSuite) TearDownSuite() {
+	gr.session.Cleanup()
+}
+
+func (gr *GlobalRolesV2TestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	gr.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(gr.T(), err)
+
+	gr.client = client
+}
+
+// Private helper function to create a Global Role with inheritedClusterRole and validate if GRBs, CRTBs, CRBs and RBs are created successfully.
+func (gr *GlobalRolesV2TestSuite) createAndTestGlobalRole(inheritedRoles []string) {
+	log.Info("Create a global role with inheritedClusterRoles.")
+	globalRole.Name = namegen.AppendRandomString("testgr")
+	globalRole.InheritedClusterRoles = inheritedRoles
+	createdGlobalRole, err := rbac.CreateGlobalRole(gr.client, localcluster, &globalRole)
+	require.NoError(gr.T(), err)
+	gr.grName = createdGlobalRole.Name
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(gr.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(gr.T(), err)
+	gr.user = createdUser
+
+	log.Info("Verify that the global role binding is created for the user.")
```

Comment:
Might be better if we separate out globalRoleCreation and validations. I noticed in one of the tests below, we are adding all the validations. Instead we can create 2 functions one for creation and one for validations of global roles v2. That way we do not have to re write the validation part in any of the tests.

Reply:
Separated global role creation and RBAC resource validation into separate functions